common/src/main/scala/net/psforever/objects/serverobject/doors/Door.scala

```diff
@@ -92,4 +122,25 @@ object Door {
     obj.Actor = context.actorOf(Props(classOf[DoorControl], obj), s"${GlobalDefinitions.door.Name}_$id")
     obj
   }
+
+  import net.psforever.types.Vector3
+  /**
+    * Instantiate and configure a `Door` object that has knowledge of both its position and outwards-facing direction.
+    * The assumption is that this door will be paired with an IFF Lock, thus, has conditions for opening.
+    * @param pos the position of the door
+    * @param outwards_direction a vector in the direction of the door's outside
+    * @param id the unique id that will be assigned to this entity
+    * @param context a context to allow the object to properly set up `ActorSystem` functionality
+    * @return the `Door` object
+    */
+  def Constructor(pos : Vector3, outwards_direction : Vector3)(id : Int, context : ActorContext) : Door = {
+    import akka.actor.Props
```

Comment:
I've been seeing imports like this all throughout the codebase. Why not import them at the beginning of the file?

Reply:
Importing resources like this limits visibility to those resources to the scope of the containing code block.  The same way one creates local variables close to the relevant place where they are used and then lets them be discardedwhen they fall out of scope, resources can be locked to specific regions where they are applicable and while they are applicable, or, in certain cases, close to where they will be referenced to reduce ambiguity of the references.

Reply:
Interesting, and since these doors are created once when the server starts, putting the `import` within the scope of the Constructor function means it could be discarded after they are all created? I'm assuming the JVM is smart enough to "release" `akka.actor.Props` if it's no longer needed, although that would not happen here because `Props` is used extensively throughout the codebase. You put it there to make it more readable.